packages/plugin-essentials/sources/commands/unlink.ts

```diff
@@ -38,38 +45,49 @@ export default class UnlinkCommand extends BaseCommand {
       throw new WorkspaceRequiredError(project.cwd, this.context.cwd);
 
     const topLevelWorkspace = project.topLevelWorkspace;
-    const unlinkedWorkspaces = new Set<string>();
+    const workspacesToUnlink = new Set<string>();
+
+    if (!this.destination) {
+      for (const {pattern, reference} of topLevelWorkspace.manifest.resolutions) {
+        if (reference.startsWith(`portal:`)) {
+          workspacesToUnlink.add(pattern.descriptor.fullName);
+        }
+      }
+    }
 
     if (this.destination) {
       const absoluteDestination = ppath.resolve(this.context.cwd, npath.toPortablePath(this.destination));
-
-      const configuration2 = await Configuration.find(absoluteDestination, this.context.plugins, {useRc: false, strict: false});
-      const {project: project2, workspace: workspace2} = await Project.find(configuration2, absoluteDestination);
-
-      if (!workspace2)
-        throw new WorkspaceRequiredError(project2.cwd, absoluteDestination);
-
-      if (this.all) {
-        for (const workspace of project2.workspaces)
-          if (workspace.manifest.name)
-            unlinkedWorkspaces.add(structUtils.stringifyIdent(workspace.locator));
-
-        if (unlinkedWorkspaces.size === 0) {
-          throw new UsageError(`No workspace found to be unlinked in the target project`);
+      if (await xfs.existsPromise(absoluteDestination)) {
```

Comment:
Went back and forth on this. Current thought: assume destination is a filepath. If the filepath doesn't exist, treat it like a glob. I don't think there's going to be some perfect "this is definitely a filepath" check we can do.

Reply:
:thinking: Perhaps we can extract and reuse [the check we already use to make `yarn <cwd> ...` (e.g. `yarn packages/yarnpkg-cli add foo` work](https://github.com/yarnpkg/berry/blob/f29676e2333e418a0ff59eb1e272788296c0f2af/packages/plugin-essentials/sources/commands/entries/run.ts#L11) - I'd personally prefer `yarn unlink jest` not to be dependent on whether the `./jest` folder exists, as users would probably use `yarn unlink ./jest` anyways to get rid of the ambiguity.

Reply:
That rule fails for globs like so:

```
{@scope/pkg-1,@scope/pkg-2}
```

since it contains a `/` but is not a valid ident. I've added a check to see if the leading argument is a valid ident, if it is, we'll process it as a glob. This way we prefer globs over filenames, and users will have to disambiguate by prefixing a leading `./`.

Reply:
Updated to use a new shared function in miscUtils.